java-generator/core/src/main/java/io/fabric8/java/generator/CRGeneratorRunner.java

```diff
@@ -20,31 +20,51 @@
 import io.fabric8.java.generator.nodes.AbstractJSONSchema2Pojo;
 import io.fabric8.java.generator.nodes.GeneratorResult;
 import io.fabric8.java.generator.nodes.JCRObject;
+import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
 import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinitionSpec;
 import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinitionVersion;
 import io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps;
-import io.fabric8.kubernetes.client.utils.Serialization;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.util.*;
 import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class CRGeneratorRunner {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(CRGeneratorRunner.class);
+
     public void run(File source, File basePath) {
         try (FileInputStream fis = new FileInputStream(source)) {
             // Parse CRD with fabric8
-            CustomResourceDefinition crd =
-                    Serialization.unmarshal(fis, CustomResourceDefinition.class);
-
-            List<WritableCRCompilationUnit> writables =
-                    generate(crd, getPackage(crd.getSpec().getGroup()));
-
-            for (WritableCRCompilationUnit w : writables) {
-                w.writeAllJavaClasses(basePath);
+            Config config = new ConfigBuilder().build();
+            KubernetesClient client = new DefaultKubernetesClient(config);
+            List<HasMetadata> resources = client.load(fis).get();
```

Comment:
You should still be able to use the Serialization class for this

Reply:
Thanks for the hint @manusa !
Can you check that the new implementation works for you?